client/app/pages/dashboards/hooks/useDuplicateDashboard.js

```diff
@@ -0,0 +1,13 @@
+import { useCallback } from "react";
+import { Dashboard } from "@/services/dashboard";
+
+export default function useDuplicateDashboard(dashboard) {
+  const duplicateDashboard = useCallback(() => {
+    const dashboardSlug = dashboard.slug;
+    Dashboard.copy({ slug: dashboardSlug }).then(({ slug }) => {
+      window.open(`dashboard/${slug}`);
+    });
```

Comment:
Please address this according to my comment in your previous PR: https://github.com/getredash/redash/pull/5232#discussion_r511338830 There is a link to similar function we use for Fork Query - just implement everything in the same way (I also explained why this is needed).

Reply:
Hi, i modefied your comment, bug i found that query and dashboard have different structures, so the lack of `geturl` method makes the current dashboard unable to reposition! So maybe someone can help me to check this problem! Thank you!